src/frontend/js/components/CourseGlimpse/index.tsx

```diff
@@ -35,6 +36,12 @@ const messages = defineMessages({
 
 const CourseGlimpseBase = ({ context, course }: CourseGlimpseProps & CommonDataProps) => {
   const intl = useIntl();
+  const organizationCoverImage = isRichieCourse(course)
+    ? course.organization_highlighted_cover_image
+    : course.organization_cover_image;
+  const organizationName = isRichieCourse(course)
+    ? course.organization_highlighted
+    : course.organization_name;
```

Comment:
As is, that's mean building a list of  `RichieCourse` in `TeacherCoursesDashboardLoader` from ou liste of `JoanieCourse`.

It may be better to duplicate theses components (glimpse and list) to have one for each entities.
The content and design of both home course list and dashboard course list may evolve in different ways.

Reply:
I've added some utils function to build CourseGlimpse props from joanie or richie course.